conda_lock/pypi_solver.py

```diff
@@ -53,16 +55,43 @@ class PlatformEnv(Env):
     Fake poetry Env to match PyPI distributions to the target conda environment
     """
 
-    def __init__(self, python_version: str, platform: str):
+    def __init__(
+        self,
+        python_version: str,
+        platform: str,
+        platform_virtual_packages: Optional[Dict[str, dict]] = None,
+    ):
         super().__init__(path=Path(sys.prefix))
         system, arch = platform.split("-")
         if arch == "64":
             arch = "x86_64"
 
         if system == "linux":
-            self._platforms = [
-                f"manylinux{tag}_{arch}" for tag in reversed(MANYLINUX_TAGS)
-            ]
+            # We use MANYLINUX_TAGS but only go up to the latest supported version
+            # as provided by __glibc if present
+            self._platforms = []
+            if platform_virtual_packages:
+                # By default, look for all tags in MANYLINUX_TAGS
+                glibc_version = MANYLINUX_TAGS[-1]
+                for p in platform_virtual_packages.values():
+                    if p["name"] == "__glibc":
+                        glibc_version = p["version"]
+                glibc_version_splits = glibc_version.split(".")
+                for tag in MANYLINUX_TAGS:
+                    if tag[0] == "_":
+                        # Compare to see if glibc_version is greater than this version
+                        if tag[1:].split("_") > glibc_version_splits:
```

Comment:
I'm concerned about string comparisons here, like `"9" < "10"` being false. We should probably use `packaging.version.Version`. (See the test I wrote below.)